src/libexpr/primops.cc

```diff
@@ -4074,54 +4093,188 @@ void EvalState::createBaseEnv()
 
     /* `builtins' must be first! */
     v.mkAttrs(buildBindings(128).finish());
-    addConstant("builtins", v);
+    addConstant("builtins", v, {
+        .type = "attribute set",
```

Comment:
```
nix-repl> builtins.typeOf {}   
"set"

```

This is going to be rendered as code, so I think

```suggestion
        .type = "set",
```

However ugly that name is, it is more correct.
Alternatively, we could render it not as code, and have a mapping from "human" types to `typeOf` type strings.

Same applies to integer/`int`, boolean/`bool` in the other constants.

Would it make sense to get the type from the `Value` instead?

Reply:
Ah I didn't know about `typeOf`. Yes we can do something better with that.

Reply:
The is a `showType` function that displays nicer strings than `builtins.typeOf`. I used it instead.